src/System.Slices/src/System/MemoryUtils.cs

```diff
@@ -0,0 +1,43 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System.Runtime.InteropServices;
+
+namespace System
+{
+    static class MemoryUtils
+    {
+        [DllImport("msvcrt.dll", CallingConvention = CallingConvention.Cdecl)]
+        private static extern unsafe int memcmp(void* first, void* second, long sizeInBytes);
+
```

Comment:
I did some tests and the original memcmp is still the fastest way to go. So I just added new MemoryUtils.Other.cs file with platform independent MemCmp implementation based on the idea from String.EqualsHelper.